src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -385,7 +385,7 @@ def compute(self, inputs, outputs):
                                     infer=False)
 
     # fill outputs with clResults
-    if clResults is not None:
+    if clResults is not None and clResults:
```

Comment:
What's the intent of the addition here? Are you trying to make sure that `clResults` is not an empty dict? It's better to check that explicitly.